bite/BiteManager.cpp

```diff
@@ -215,60 +214,75 @@ ptr<vector<ptr<AESKeyDecryptionShare> > > BiteManager::getDecryptionSharesFromDa
 }
 
 
-ptr<DecryptedTransactionDataFields> BiteManager::verifyAndDecryptTransactionList(
+ptr<DecryptedTransactionFieldsMap> BiteManager::verifyAndDecryptTransactionList(
     TransactionList &_transactionList, DecryptedAESKeyList &_aesKeys) {
-    auto decryptedDataFields = make_shared<DecryptedTransactionDataFields>();
+    auto decryptedFieldsMap = make_shared<DecryptedTransactionFieldsMap>();
 
     auto txs = _transactionList.getItems();
     CHECK_STATE(txs);
 
     try {
         for (uint64_t i = 0; i < _transactionList.size(); i++) {
             auto tx = txs->at(i);
-            auto bite = tx->parseAndValidateBiteDataField();
+            tx->parseAndValidate();
+            auto bite = tx->tryGetBiteData();
             if (bite) {
                 auto decryptedAESKey = _aesKeys.getKey(i);
                 CHECK_STATE(decryptedAESKey);
 
-                ptr<vector<uint8_t> > decryptedOriginalDataField = nullptr;
-
                 try {
-                    decryptedOriginalDataField = decryptDataField(bite, *decryptedAESKey);
+                    auto decryptedTransactionFields = decryptDataField(bite, *decryptedAESKey);
+                    decryptedFieldsMap->emplace(i, decryptedTransactionFields);
                 } catch (const std::exception &e) {
                     LOG(err, fmt::format("Corrupt tx:{} that doesnt decrypt: {}", i, e.what()));
-                    decryptedDataFields->emplace(i, nullptr);;
```

Comment:
if a transaction couldn't be decrypted it should still be added to the list and passed to skaled. skaled then charges the users who submitted invalid transaction on purpose